sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -1689,6 +1689,11 @@ public WriteResult expand(PCollection<T> input) {
         if (getMaxFileSize() != null) {
           batchLoads.setMaxFileSize(getMaxFileSize());
         }
+        // When running in streaming (unbounded mode) we want to retry failed load jobs
+        // indefinitely. Failing the bundle is expensive, so we set a fairly high limit on retries.
+        if (IsBounded.UNBOUNDED.equals(input.isBounded())) {
+          batchLoads.setMaxRetryJobs(1000);
```

Comment:
BQ daily jobs quota is 1000. So 1000 retries seems to be unrealistic.

Reply:
Failed jobs will retry forever regardless of this setting, and streaming jobs retry forever. Without this the entire bundle is retried (because runners retry bundles not elements), and potentially unrelated loads are retried which is even worse for daily job quota.

Reply:
Also, I believe that BQ's daily job per table per day quota does not count failed jobs. Only jobs that will complete successfully consume this quota.

Reply:
How difficult would it be to make them not retry forever in case we want to route them to FailedInserts? I'm investigating https://issues.apache.org/jira/browse/BEAM-7693 and this seems to be implicated. Thanks and sorry for the 🧟‍♂️thread revival :)